addons/html_builder/static/src/sidebar/block_tab.js

```diff
@@ -134,4 +153,223 @@ export class BlockTab extends Component {
     }
 
     // TODO bounce animation on click if empty editable
+
+    makeSnippetDraggable() {
+        let dropzoneEls = [];
+        let dragAndDropResolve;
+
+        let snippet, snippetEl, isSnippetGroup;
+
+        const iframeWindow =
+            this.document.defaultView !== window ? this.document.defaultView : false;
+
+        const scrollingElement = () =>
+            this.shared.dropzone.getDropRootElement() ||
+            this.editable.querySelector(".o_notebook") ||
+            getScrollingElement(this.document) ||
+            this.editable.querySelector(".o_editable");
+
+        const dragAndDropOptions = {
+            ref: { el: this.blockTabRef.el },
+            iframeWindow,
+            cursor: "move",
+            el: this.blockTabRef.el,
+            elements: ".o_snippet.o_draggable",
+            scrollingElement,
+            handle: ".o_snippet_thumbnail:not(.o_we_ongoing_insertion .o_snippet_thumbnail)",
+            dropzones: () => dropzoneEls,
+            helper: ({ element, helperOffset }) => {
+                snippet = element;
+                const draggedEl = element.cloneNode(true);
+                draggedEl
+                    .querySelectorAll(
+                        ".o_snippet_thumbnail_title, .o_snippet_thumbnail_area, .rename-delete-buttons"
+                    )
+                    .forEach((el) => el.remove());
+                draggedEl.style.position = "fixed";
+                document.body.append(draggedEl);
+                // Center the helper on the thumbnail image.
+                const thumbnailImgEl = element.querySelector(".o_snippet_thumbnail_img");
+                helperOffset.x = thumbnailImgEl.offsetWidth / 2;
+                helperOffset.y = thumbnailImgEl.offsetHeight / 2;
+                return draggedEl;
+            },
+            onDragStart: ({ element }) => {
+                this.cancelDragAndDrop = this.shared.history.makeSavePoint();
+                this.hideSnippetToolTip?.();
+
+                this.shared.operation.next(async () => {
+                    await new Promise((resolve) => (dragAndDropResolve = () => resolve()));
```

Comment:
you can use a deferred instead (from web/utils/concurrency). maybe it would be clearer